File: data/readable_format/fortran/file270.f90
module math_operations_mod
    implicit none
contains
    subroutine updateY(y, ed, ek, tiltc, tilts, crkve, cikve, one, dpda, ix, i)
        double precision, intent(inout) :: y(2)
        double precision, intent(in) :: ed(*), ek(*), tiltc(*), tilts(*)
        double precision, intent(in) :: crkve, cikve, one, dpda
        integer, intent(in) :: ix, i

        y(1) = y(1) + (ed(ix) * tiltc(i) * crkve - ek(ix) * tilts(i) * cikve) / (one + dpda)
        y(2) = y(2) + (ek(ix) * tiltc(i) * cikve + ed(ix) * tilts(i) * crkve) / (one + dpda)
    end subroutine updateY
end module math_operations_mod

program test
    use math_operations_mod
    implicit none

    double precision, dimension(2) :: y
    double precision, dimension(5) :: ed, ek, tiltc, tilts
    double precision :: crkve, cikve, one, dpda
    integer :: ix, i

    ! Initialize some test values
    y = (/0.0, 0.0/)
    ed = (/1.0, 2.0, 3.0, 4.0, 5.0/)
    ek = (/5.0, 4.0, 3.0, 2.0, 1.0/)
    tiltc = (/0.1, 0.2, 0.3, 0.4, 0.5/)
    tilts = (/0.5, 0.4, 0.3, 0.2, 0.1/)
    crkve = 0.6
    cikve = 0.7
    one = 1.0
    dpda = 0.5
    ix = 3
    i = 3

    ! Perform the operation
    call updateY(y, ed, ek, tiltc, tilts, crkve, cikve, one, dpda, ix, i)

    ! Print the result for verification
    print *, "Y(1) = ", y(1)
    print *, "Y(2) = ", y(2)
end program test

File: data/readable_format/cpp/file270.cpp
#include <vector>
#include <iostream>
#include <cmath> // Included for std::abs

void updateY(std::vector<double>& y, const std::vector<double>& ed, const std::vector<double>& ek,
             const std::vector<double>& tiltc, const std::vector<double>& tilts,
             double crkve, double cikve, double one, double dpda, int ix, int i) {
    y[0] = y[0] + (ed[ix-1] * tiltc[i-1] * crkve - ek[ix-1] * tilts[i-1] * cikve) / (one + dpda);
    y[1] = y[1] + (ek[ix-1] * tiltc[i-1] * cikve + ed[ix-1] * tilts[i-1] * crkve) / (one + dpda);
}

// A simple assertion function to check if the test passes
void assertClose(double actual, double expected, double tolerance = 1e-5) {
    if (std::abs(actual - expected) > tolerance) {
        std::cerr << "Test failed: Expected " << expected << ", got " << actual << std::endl;
        exit(EXIT_FAILURE);
    }
}

int main() {
    // Test updateY function with the same inputs as the Fortran code
    std::vector<double> y = {0.0, 0.0};
    std::vector<double> ed = {1.0, 2.0, 3.0, 4.0, 5.0};
    std::vector<double> ek = {5.0, 4.0, 3.0, 2.0, 1.0};
    std::vector<double> tiltc = {0.1, 0.2, 0.3, 0.4, 0.5};
    std::vector<double> tilts = {0.5, 0.4, 0.3, 0.2, 0.1};
    double crkve = 0.6;
    double cikve = 0.7;
    double one = 1.0;
    double dpda = 0.5;
    int ix = 3;
    int i = 3;

    updateY(y, ed, ek, tiltc, tilts, crkve, cikve, one, dpda, ix, i);

    // Output the results for verification
    std::cout << "Y[0] = " << y[0] << std::endl;
    std::cout << "Y[1] = " << y[1] << std::endl;

    // Assertions to verify the correctness
    assertClose(y[0], -0.059999980926512819); // Expected value for y[0] based on Fortran output
    assertClose(y[1], 0.78000003814697294);   // Expected value for y[1] based on Fortran output

    std::cout << "All tests passed." << std::endl;

    return 0;
}